src/common/function/FunctionManager.cpp

```diff
@@ -577,10 +582,78 @@ FunctionManager::FunctionManager() {
         }
         case Value::Type::INT:
         case Value::Type::FLOAT: {
-          if (args.size() == 2) {
+          if (args.size() >= 2) {
             if (args[1].get().type() == Value::Type::INT) {
+              auto val = args[0].get().getFloat();
               auto decimal = args[1].get().getInt();
-              return std::round(args[0].get().getFloat() * pow(10, decimal)) / pow(10, decimal);
+              auto factor = pow(10.0, decimal);
+
+              string mode = "half_up";
+              if (args.size() == 3 && args[2].get().type() == Value::Type::STRING) {
+                mode = args[2].get().getStr();
+              }
+              if (mode == "up") {
```

Comment:
Make the rounding mode case insensitive.

Reply:
I adopted boost:iequal to make the rounding mode comparison case insensitive.